src/env.rs

```diff
@@ -74,50 +71,44 @@ impl Environment {
     /// It requires you to provide the `EventQueue` as well because
     /// the initialization process does a few roundtrip to the server
     /// to initialize all the globals.
-    ///
-    /// This may panic or fail if you do not provide the EventQueue hosting
-    /// the registry you provided.
-    pub fn from_registry(
-        registry: NewProxy<wl_registry::WlRegistry>,
+    pub fn from_display(
+        display: &Proxy<wl_display::WlDisplay>,
         evq: &mut EventQueue,
     ) -> io::Result<Environment> {
-        Environment::from_registry_with_cb(registry, evq, |_, _| {})
+        Environment::from_display_with_cb(display, evq, |_, _| {})
     }
 
     /// Create an environment wrapping a new registry
     ///
-    /// Additionnaly to `from_registry`, this allows you to provide
+    /// Additionnaly to `from_display`, this allows you to provide
     /// a callback to be notified of global events, just like
     /// `GlobalManager::new_with_cb`. Note that you will still
     /// receive events even if they are processed by this `Environment`.
-    pub fn from_registry_with_cb<Impl>(
-        registry: NewProxy<wl_registry::WlRegistry>,
+    pub fn from_display_with_cb<Impl>(
+        display: &Proxy<wl_display::WlDisplay>,
         evq: &mut EventQueue,
         mut cb: Impl,
     ) -> io::Result<Environment>
     where
-        Impl: Implementation<Proxy<wl_registry::WlRegistry>, GlobalEvent> + Send + 'static,
+        Impl: FnMut(GlobalEvent, Proxy<wl_registry::WlRegistry>) + Send + 'static,
     {
         let outputs = ::output::OutputMgr::new();
         let outputs2 = outputs.clone();
 
-        let manager = GlobalManager::new_with_cb(registry, move |event, registry: Proxy<_>| {
+        let manager = GlobalManager::new_with_cb(&display, move |event, registry| {
```

Comment:
Actually, for this to be correct now that we create the registry ourselves, we need to ensure that the registry will be created on the event queue that was given as argument.

To ensure that, we need to first make a wrapper of the display:

```rust
let display_wrapper = display.make_wrapper(&evq.get_token());
```

and then give this wrapper to `GlobalManager::new_with_cb(..)`